src/Bicep.Core/Semantics/Decorator.cs

```diff
@@ -42,6 +44,8 @@ public Decorator(FunctionOverload overload, TypeSymbol attachableType, Decorator
 
         public void Validate(DecoratorSyntax decoratorSyntax, TypeSymbol targetType, ITypeManager typeManager, IBinder binder, IDiagnosticWriter diagnosticWriter)
         {
+            targetType = RemoveImplicitNull(targetType);
```

Comment:
I'm not sure if this is the right approach. All existing ARM constraints (except `type`) are no-ops if invoked on a value of `null`, but extensibility providers may have a need to block usage of an unforeseen decorator on nullable params or types. 

The alternative is to update all the decorator definitions with assignability restrictions in SystemNamespaceType so that they accept nullable types (e.g., `@minValue` would need to be assignable to `int | null` instead of just `int`). This might make us more likely to accidentally block a future decorator's usage on nullable params, though, and is more verbose.

Reply:
It's a little quirky, but in the interest of not changing existing behavior, and with the fact that decorators are very infrequently added, I think it's OK - as long as you explain this well in a comment to avoid future surprises.